### PR TITLE
Add build_vm_compilers package with support for modularly compiling vm apps 

### PR DESCRIPTION
**Pull request by jakemac53**
Towards https://github.com/dart-lang/build/issues/1428

### CODE REVIEW

File: build_modules/lib/src/kernel_builder.dart [hunk 1]
```diff
@@ -17,83 +18,121 @@ import 'modules.dart';
...
+/// This creates kernel files based on [moduleExtension] files, which are what
+/// determine the module structure of an application.
+class KernelBuilder implements Builder {
   @override
-  final buildExtensions = const {
-    moduleExtension: const [kernelSummaryExtension]
-  };
+  final Map<String, List<String>> buildExtensions;
+
+  /// The output extension for this builder.
```
Comment (natebosch) [inline build_modules/lib/src/kernel_builder.dart]: [nit] drop this comment

**Comment by jakemac53** [comment]
Note that this will need https://dart-review.googlesource.com/c/sdk/+/62120 before the test can actually pass

File: build_modules/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,9 @@
+## 0.3.0-dev
+
+- Removed the `kernelSummaryExtension`, and renamed the `KernelSummaryBuilder`
```
Comment (natebosch) [inline build_modules/CHANGELOG.md]: can you make it more clear how this is breaking?

Comment (jakemac53) [reply build_modules/CHANGELOG.md]: How so? The only breaking things are the two items described here...

File: build_modules/lib/src/kernel_builder.dart (hunk 1 above)
Comment (jakemac53) [reply build_modules/lib/src/kernel_builder.dart]: done

File: build_modules/CHANGELOG.md (hunk 2 above)
Comment (natebosch) [reply build_modules/CHANGELOG.md]: Fair enough

**Comment by jakemac53** [comment]
Going to wait on this until the next sdk drops, which should be soonish?

**Pull request merged**